lua/entities/wtib_tiberium_base/init.lua

```diff
@@ -226,23 +210,23 @@ function ENT:SetField(num)
 	self:SetTiberiumFieldID(num)
 end
 
-function ENT:SetTiberiumAmount(am)
-
-	if am <= 0 then self:Die() return end
-	
-	self:SetTiberiumAmount(math.Clamp(am,1,self:GetMaxTiberiumAmount()))
-	
+function ENT:AddTiberiumAmount(am)
+	self:SetTiberiumAmount(math.Clamp(self:GetTiberiumAmount() + am,0,self:GetMaxTiberiumAmount()))
 	if self.NextReproduce <= CurTime() and self:GetTiberiumAmount() >= self.Reproduce_TiberiumRequired then self:AttemptReproduce() end // Check if we should reproduce
-	
 	self:CalcSize()
-	self:CheckColor()
-	
 end
 
-function ENT:AddTiberiumAmount(am)
-	self:SetTiberiumAmount(self:GetTiberiumAmount() + am)
-end
+local TibSize = 1
+local LastTibSize = TibSize
 
 function ENT:DrainTiberiumAmount(am)
```

Comment:
That is actually used for something i changed.  It is sued below in the math.approach which makes the crystal shrink smoothly instead of just vaporizing as a large crystal when its all harvested out.